overlord/exportstate/exportmgr.go

```diff
@@ -0,0 +1,137 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"sync"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/timings"
+)
+
+// ExportManager is responsible for maintenance of content exported from snaps
+// to other snaps or to the host system and, in some cases, for content exported
+// from the host system to snaps.
+//
+// The export manager stores state describing the content exported by each
+// particular revision. Content exported by the host is a special case and is
+// not stored in the state.
```

Comment:
thanks for covering the use cases in the description here. Some of this should also go in the package doc comment.  See assertstate vs AssertManager for example.

Reply:
I tweaked the package description, have a look please.